edge-modules/MetricsCollector/Program.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace MetricsCollector
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+    using Microsoft.Azure.Devices.Edge.Agent.Diagnostics;
+    using Microsoft.Azure.Devices.Edge.Agent.Diagnostics.Publisher;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    internal class Program
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("MetricsCollector");
+
+        public static int Main() => MainAsync().Result;
+
+        private static async Task<int> MainAsync()
+        {
+            Logger.LogInformation($"Starting metrics collector with the following settings:\r\n{Settings.Current}");
+
+            MetricsScraper scraper = new MetricsScraper(Settings.Current.Endpoints);
+            IMetricsPublisher publisher;
+            if (Settings.Current.UploadTarget == UploadTarget.AzureLogAnalytics)
+            {
+                publisher = new LogAnalyticsUpload(Settings.Current.AzMonWorkspaceId, Settings.Current.AzMonWorkspaceKey, Settings.Current.AzMonLogType, Guid.NewGuid());
+            }
+            else
+            {
+                MqttTransportSettings mqttSetting = new MqttTransportSettings(TransportType.Mqtt_Tcp_Only);
+                ITransportSettings[] transportSettings = { mqttSetting };
+                ModuleClient moduleClient = await ModuleClient.CreateFromEnvironmentAsync(transportSettings);
+                publisher = new IoTHubMetricsUpload(moduleClient);
+            }
+
+            MetricsScrapeAndUpload metricsScrapeAndUpload = new MetricsScrapeAndUpload(scraper, publisher);

```

Comment:
since this is an IDisposable, it should be disposed before shutdown. I recommend a ```using``` statement